invenio_rdm_records/services/generators.py

```diff
@@ -262,3 +264,21 @@ def __init__(self, config_key, accept_values=[True], **kwargs):
     def _condition(self, **_):
         """Check if the config value is truthy."""
         return current_app.config.get(self.config_key) in self.accept_values
+
+
+class ResourceAccessToken(RecordOwners):
+    """Allow resource access tokens."""
+
+    def __init__(self, access):
+        """Constructor."""
+        self.access = access
+
+    def needs(self, record=None, file_key=None, **kwargs):
+        """Enabling Needs."""
+        record_owner = super().needs(record)
+
+        if record_owner and record and file_key:
+            signer_id = record_owner[0].value
```

Comment:
```suggestion
class ResourceAccessToken(Generator):
    """Allow resource access tokens."""

    def __init__(self, access):
        """Constructor."""
        self.access = access

    def needs(self, record=None, file_key=None, **kwargs):
        """Enabling Needs."""
        record_owner = next(iter(record.parent.access.owners), None)

        if record_owner and record and file_key:
            signer_id = record_owner
```

nit: in the end, it looks a bit awkward to get the needs from the `RecordOwners` generator, and then "extract" the user ID from them to use again here, so duplicating might be simpler and more readable. 

The alternative would be to extract the logic in a `RecordOwners.get_owners(self, record)` method, but it's a one-liner, so it feels a bit too much.